### ISSUE TITLE
Tests fail for no reason

### ISSUE THREAD

**Issue by netoneko** [logstash-plugins/logstash-input-s3#42]
`bundle exec rspec` returns 13 failures on master branch and also on old tags. Am I missing something when I run it like that?

Example output (another 11 tests fail too):

```
Failures:

  1) LogStash::Inputs::S3#list_new_files should sort return object sorted by last_modification date with older first
     Failure/Error: expect(config.list_new_files).to eq(['TWO_DAYS_AGO', 'YESTERDAY', 'TODAY'])

       expected: ["TWO_DAYS_AGO", "YESTERDAY", "TODAY"]
            got: ["TODAY"]

       (compared using ==)
     # ./spec/inputs/s3_spec.rb:124:in `(root)'

  2) LogStash::Inputs::S3#list_new_files should support not providing a exclude pattern
     Failure/Error: expect(config.list_new_files).to eq(objects_list.map(&:key))

       expected: ["exclude-this-file-1", "exclude/logstash", "this-should-be-present"]
            got: ["this-should-be-present"]

       (compared using ==)
     # ./spec/inputs/s3_spec.rb:69:in `(root)'
```

**Comment by ph** [comment]
Weird, I have encountered this issue today, but the test are running the CI

http://build-eu-1.elasticsearch.org/job/logstash-plugin-input-s3-unit/

**Comment by ph** [comment]
This look related to a `~/.sincedb_*` file present in the home directory that could make the test fails for some reason. The suite should not use the .sincedb files in $HOME.

**Comment by ph** [comment]
Fixed in #55
